worker-core/src/main/java/com/github/workerframework/core/TrackingMessageCreatorImpl.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2015-2025 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.workerframework.core;
+
+import com.github.cafapi.common.api.Codec;
+import com.github.cafapi.common.api.CodecException;
+import com.github.workerframework.api.TaskMessage;
+import com.github.workerframework.api.TaskStatus;
+import com.github.workerframework.api.TrackingMessageCreator;
+import com.github.workerframework.tracking.report.TrackingReport;
+import com.github.workerframework.tracking.report.TrackingReportConstants;
+import com.github.workerframework.tracking.report.TrackingReportFailure;
+import com.github.workerframework.tracking.report.TrackingReportStatus;
+import com.github.workerframework.tracking.report.TrackingReportTask;
+import com.github.workerframework.util.rabbitmq.RabbitHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.github.workerframework.core.WorkerTaskImpl.getWorkerName;
+
+public final class TrackingMessageCreatorImpl implements TrackingMessageCreator {
```

Comment:
Make this a singleton

```suggestion
public final enum TrackingMessageCreatorImpl implements TrackingMessageCreator {

    INSTANCE;
```